build/lib/src/analyzer/resolver.dart

```diff
@@ -14,29 +14,30 @@ abstract class Resolver {
   ///
   /// This will be `false` in the case where the file is not Dart source code,
   /// or is a `part of` file (not a standalone Dart library).
-  bool isLibrary(AssetId assetId);
+  Future<bool> isLibrary(AssetId assetId);
 
   /// All libraries accessible from the entry point, recursively.
   ///
   /// **NOTE**: This includes all Dart SDK libraries as well.
-  Iterable<LibraryElement> get libraries;
+  Stream<LibraryElement> get libraries;
 
   /// Returns a resolved library representing the file defined in [assetId].
   ///
   /// * Throws [NonLibraryAssetException] if [assetId] is not a Dart library.
-  LibraryElement getLibrary(AssetId assetId);
-
-  /// Finds the first library identified by [libraryName], or null if no
-  /// library can be found.
+  Future<LibraryElement> libraryFor(AssetId assetId);
+  @Deprecated('Use libraryFor')
+  Future<LibraryElement> getLibrary(AssetId assetId);
```

Comment:
I get that `get*` methods are redundant/annoying but I am not really sure if its worth a breaking change? We don't save any typing (actually add a character below) or provide any additional value with this change.